libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -326,12 +337,38 @@ bool AC_Fence::pre_arm_check(const char* &fail_msg) const
         return false;
     }
 
-    if (_alt_max < _alt_min) {
+#if AP_TERRAIN_AVAILABLE
+    const auto *terrain = AP::terrain();
+    if (Location::AltFrame(_alt_frame.get()) == Location::AltFrame::ABOVE_TERRAIN &&
```

Comment:
```suggestion
    if (_alt_frame == Location::AltFrame::ABOVE_TERRAIN &&
```

Use `AP_Enum<Location::AltFrame>` for the _alt_frame instance variable to make all of this much more readable.